test/unit_tests/braket/experimental/autoqasm/test_parameters.py

```diff
@@ -412,11 +463,11 @@ def cal_1(angle: float):
         pulse.delay("$1", angle)
 
     @aq.main
-    def my_program():
-        rx("$1", FreeParameter("theta"))
+    def my_program(theta):
+        rx("$1", theta)
 
-    qasm1 = my_program().with_calibrations(cal_1).make_bound_program({"theta": 0.6}).to_ir()
-    qasm2 = my_program().make_bound_program({"theta": 0.6}).with_calibrations(cal_1).to_ir()
+    qasm1 = my_program.with_calibrations(cal_1).make_bound_program({"theta": 0.6}).to_ir()
+    qasm2 = my_program.make_bound_program({"theta": 0.6}).with_calibrations(cal_1).to_ir()
```

Comment:
This `make_bound_progarm` API doesn't make sense anymore. Let's update with `my_program(theta=0.6)`, like the `Circuit` class has

Reply:
This also is halfway towards fixing one of the failing tests -- it gets rid of the TypeError, but we still have to find a new way to raise an error when `main` calls `main`

Reply:
As discussed offline and in other comments, the program `__call__` method may yet be used for other purposes, like running the program. This use of `make_bound_program` feels pretty similar to how it was before this change, the IR isn't modified at all; the only thing that's changed is how the FP is declared.

In general, definitely open to finding a smoother interface here though